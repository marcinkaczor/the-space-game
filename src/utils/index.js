import * as PLAYERS from '../constants/players'

const capitalizeFirstLetter = string => string.charAt(0).toUpperCase() + string.slice(1)

const randomItemFromArray = array => array[Math.floor(Math.random() * array.length)]

const getFirstChar = string => string.charAt(0)

const increaseScore = player => score => {
  if (player === PLAYERS.LEFT) return `${Number(score.split('-')[0]) + 1}-${Number(score.split('-')[1])}`
  if (player === PLAYERS.RIGHT) return `${Number(score.split('-')[0])}-${Number(score.split('-')[1]) + 1}`

  return score
}

export {
  capitalizeFirstLetter as capitalize,
  randomItemFromArray as randomItem,
  getFirstChar,
  increaseScore
}
