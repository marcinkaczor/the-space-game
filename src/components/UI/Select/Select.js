import React from 'react'
import PropTypes from 'prop-types'
import { FormControl, InputLabel, Select, MenuItem, FormHelperText } from '@material-ui/core'

import { capitalize } from '../../../utils'

const StellarSelect = ({ label, helper, values, currentValue, setCurrentValue }) => (
  <FormControl>
    <InputLabel id={`select-${label.trim()}-label`}>{capitalize(label)}</InputLabel>
    <Select
      labelId={`select-${label.trim()}-label`}
      id={`select-${label.trim()}`}
      value={currentValue}
      onChange={e => setCurrentValue(e.target.value)}
    >
      {values.map((value, i) => <MenuItem key={i} value={value}>{capitalize(value)}</MenuItem>)}
    </Select>
    {helper && <FormHelperText>{capitalize(helper)}</FormHelperText>}
  </FormControl>
)

StellarSelect.propTypes = {
  label: PropTypes.string.isRequired,
  helper: PropTypes.string,
  values: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentValue: PropTypes.string.isRequired,
  setCurrentValue: PropTypes.func.isRequired
}

export default StellarSelect
