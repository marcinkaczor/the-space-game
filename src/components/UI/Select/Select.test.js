/* global describe, test, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import { capitalize } from '../../../utils'

import Select from '.'

const label = 'Label'
const helper = 'Select which value you want to choose'
const values = ['foo', 'bar']
const currentValue = 'foo'
const setCurrentValue = () => {}

describe('Select', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <Select
        label={label}
        values={values}
        currentValue={currentValue}
        setCurrentValue={setCurrentValue}
      />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with its props', () => {
    const wrapper = mount(
      <Select
        label={label}
        helper={helper}
        values={values}
        currentValue={currentValue}
        setCurrentValue={setCurrentValue}
      />
    )
    expect(wrapper.find('label').text()).toEqual(capitalize(label))
    expect(wrapper.find('p').text()).toEqual(capitalize(helper))
    expect(wrapper.find('input').instance().value).toEqual(currentValue)
  })
})
