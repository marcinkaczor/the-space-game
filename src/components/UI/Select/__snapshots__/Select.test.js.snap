// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select snapshot renders 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="select-hello-label"
  >
    Hello
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="select-hello-label select-hello"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="select-hello"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Foo
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
